### PR TITLE
Fix client name

### PR DESCRIPTION
**Pull request by qiaozha**
we have been using getLibraryName from tcgc for clientName, however, it's not necessary as tcgc has already handled the `@clientName` for us if it's being used at the client.

### CODE REVIEW

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 1]
```diff
@@ -1747,21 +1748,17 @@ function getMethodApiVersionParameter(): Parameter | void {
...
-  namespace: string,
   rlcModelsMap: Map<string, RLCModel>
 ): HrlcClient[] {
   const program = context.program;
   const clients = listClients(context);
   const retval: HrlcClient[] = [];
   methodApiVersionParam = undefined;
   for (const client of clients) {
-    const clientName = getLibraryName(context, client.type).replace(
+    const clientName = client.name.replace(
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:1758]: tcgc has already handled the client name if it has `@clientName` decorator.

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 2]
```diff
@@ -1859,26 +1855,14 @@ export function emitCodeModel(
...
+      codeModel.clients = emitClients(dpgContext, rlcModelsMap);
       codeModel.clients.length > 1 &&
         codeModel.clients.map((client) => {
           client["subfolder"] = normalizeName(
             client.name.replace("Client", ""),
             NameType.File
           );
         });
-    } else {
-      codeModel["subnamespaceToClients"][namespace] = emitClients(
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:1871]: this is not used now.

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 3]
```diff
@@ -1888,4 +1872,4 @@ export function emitCodeModel(
     ...simpleTypesMap.values()
   ];
   return codeModel;
-}
+}
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:1875]:
```suggestion
}

```

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 4]
```diff
@@ -1747,21 +1748,14 @@ function getMethodApiVersionParameter(): Parameter | void {
...
   methodApiVersionParam = undefined;
   for (const client of clients) {
-    const clientName = getLibraryName(context, client.type).replace(
-      "Client",
-      ""
-    );
-    if (getNamespace(context, client.name) !== namespace) {
-      continue;
-    }
+    const clientName = client.name.replace("Client", "");
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:1758]: is this the change?

**Review by MaryGao — APPROVED** [review]

**Pull request merged**